isis/tests/CSMCameraTests.cpp

```diff
@@ -314,24 +460,298 @@ TEST_F(CSMCameraFixture, Resolution) {
+      .WillOnce(::testing::Return(imageLocus.point));
 
-  // We also have to set the mock up for setimage
-  EXPECT_CALL(mockModel, imageToRemoteImagingLocus(MatchImageCoord(imagePt), ::testing::_, ::testing::_, ::testing::_))
+  double position[3];
+  testCam->instrumentBodyFixedPosition(position);
+  EXPECT_EQ(position[0], (imageLocus.point.x) / 1000.0);
+  EXPECT_EQ(position[1], (imageLocus.point.y) / 1000.0);
+  EXPECT_EQ(position[2], (imageLocus.point.z) / 1000.0);
+}
+
+
+TEST_F(CSMSetCameraFixture, SubSpacecraftPoint) {
+  EXPECT_CALL(mockModel, getSensorPosition(MatchImageCoord(imagePt)))
       .Times(1)
-      .WillOnce(::testing::Return(csm::EcefLocus(3394200.43980104 + 50000, 0, 0, -1, 0, 0)));
-  EXPECT_CALL(mockModel, computeGroundPartials)
-      .Times(2)
-      .WillRepeatedly(::testing::Return(std::vector<double>{1, 2, 3, 4, 5, 6}));
+      .WillOnce(::testing::Return(imageLocus.point));
 
-  testCam->SetImage(5, 5);
   double lat, lon;
   testCam->subSpacecraftPoint(lat, lon);
   EXPECT_EQ(lat, 0.0);
   EXPECT_EQ(lon, 0.0);
+}
+
+
+TEST_F(CSMSetCameraFixture, SlantDistance) {
+  EXPECT_CALL(mockModel, getSensorPosition(MatchImageCoord(imagePt)))
+      .Times(1)
+      .WillOnce(::testing::Return(imageLocus.point));
+
+  double expectedDistance = sqrt(
+      pow(imageLocus.point.x - groundPt.x, 2) +
+      pow(imageLocus.point.y - groundPt.y, 2) +
+      pow(imageLocus.point.z - groundPt.z, 2)) / 1000.0;
+  EXPECT_DOUBLE_EQ(testCam->SlantDistance(), expectedDistance);
+}
+
+
+TEST_F(CSMSetCameraFixture, TargetCenterDistance) {
+  EXPECT_CALL(mockModel, getSensorPosition(MatchImageCoord(imagePt)))
+      .Times(1)
+      .WillOnce(::testing::Return(imageLocus.point));
+
+  double expectedDistance = sqrt(
+      pow(imageLocus.point.x, 2) +
+      pow(imageLocus.point.y, 2) +
+      pow(imageLocus.point.z, 2)) / 1000.0;
+  EXPECT_DOUBLE_EQ(testCam->targetCenterDistance(), expectedDistance);
+}
+
+
+TEST_F(CSMSetCameraFixture, PhaseAngle) {
+  EXPECT_CALL(mockModel, getSensorPosition(MatchImageCoord(imagePt)))
+      .Times(1)
+      .WillOnce(::testing::Return(csm::EcefCoord(groundPt.x + 50000, groundPt.y, groundPt.z + 50000)));
+  EXPECT_CALL(mockModel, getIlluminationDirection(MatchEcefCoord(groundPt)))
+      .Times(1)
+      .WillOnce(::testing::Return(csm::EcefVector(0.0, 0.0, 1.0)));
+
+  EXPECT_DOUBLE_EQ(testCam->PhaseAngle(), 45.0);
+}
+
+
+TEST_F(CSMSetCameraFixture, IncidenceAngle) {
+  EXPECT_CALL(mockModel, getIlluminationDirection(MatchEcefCoord(groundPt)))
+      .Times(1)
+      .WillOnce(::testing::Return(csm::EcefVector(0.0, 0.0, 1.0)));
+
+  EXPECT_DOUBLE_EQ(testCam->IncidenceAngle(), 90.0);
+}
+
+
+TEST_F(CSMSetCameraFixture, EmissionAngle) {
+  EXPECT_CALL(mockModel, getSensorPosition(MatchImageCoord(imagePt)))
+      .Times(1)
+      .WillOnce(::testing::Return(imageLocus.point));
+
+  EXPECT_DOUBLE_EQ(testCam->EmissionAngle(), 0.0);
+}
+
+
+TEST_F(CSMCameraFixture, SetTime) {
```

Comment:
I hate how much boilerplate these tests take, but that's a problem for another day.

Reply:
It does take a lot just to get the test going